cmd/apps/cockroachdb_app.go

```diff
@@ -0,0 +1,100 @@
+// Copyright (c) arkade author(s) 2021. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+package apps
+
+import (
+	"fmt"
+	"github.com/alexellis/arkade/pkg"
+	"github.com/alexellis/arkade/pkg/apps"
+	"github.com/alexellis/arkade/pkg/config"
+	"github.com/alexellis/arkade/pkg/types"
+	"github.com/spf13/cobra"
+)
+
+func MakeInstallCockroachdb() *cobra.Command {
+	var cockroachdb = &cobra.Command{
+		Use:          "cockroachdb",
+		Short:        "Install cockroachdb",
+		Long:         "Install cockroachdb",
+		Example:      ` arkade app install cockroachdb`,
+		SilenceUsage: true,
+	}
+	cockroachdb.Flags().String("namespace", "default", "Namespace for the app")
+	cockroachdb.Flags().Bool("persistence", false, "Use a 100Gi Persistent Volume to store data")
+
+	cockroachdb.Flags().Bool("single-node", false, "Run CockroachDB instances in standalone mode with replication disabled, so the StatefulSet does NOT FORM A CLUSTER")
+	cockroachdb.Flags().Int64("replicas", 1, "Satetfulset replica count")
+
+	cockroachdb.Flags().Bool("enable-tls", false, "Whether to run securely using TLS certificates")
+
+	cockroachdb.Flags().StringArray("set", []string{}, "Use custom flags or override existing flags \n(example --set tls.enabled=false)")
+
+	cockroachdb.RunE = func(command *cobra.Command, args []string) error {
+		kubeConfigPath, _ := command.Flags().GetString("kubeconfig")
+		if err := config.SetKubeconfig(kubeConfigPath); err != nil {
+			return err
+		}
+
+		namespace, _ := command.Flags().GetString("namespace")
+		persistence, _ := command.Flags().GetBool("persistence")
+		singleNode, _ := command.Flags().GetBool("single-node")
+		enableTls, _ := command.Flags().GetBool("enable-tls")
+		replicas, _ := command.Flags().GetInt64("replicas")
+
+		overrides := map[string]string{}
+
+		if singleNode {
+			overrides["conf.single-node"] = "true"
+		}
+		if enableTls {
+			overrides["tls.enabled"] = "false"
```

Comment:
Does this not require additional settings like Ingress being turned on, or does it work through a TCP connection?

Is it a self-signed cert?

Reply:
This is TLS for the database connection itself, not for the dashboard.
It is self-signed, but has options in the values to provide one, so in this case you can provide it with the `--set` flag.

Reply:
How does the user download the TLS secret and put it into their trust bundle? Do we need a command for that?

Reply:
If TLS is enabled, you should download the CA cert and you should put it into your `.postgresql` folder in your application user's home directory, and you should use the client certificate in your application to connect to the DB, so in case of golang your should put it into your `pgx` connection.

From my perspective, this message is all about how to install and manage the installed application from the devops side.
I don't think this is in the scope of installing or managing the cluster, more like how do you use it in code.
Am I wrong in this case?